gradle-plugin-testing-junit/src/main/java/com/palantir/gradle/testing/execution/InvocationResultAssert.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.testing.execution;
+
+import java.util.Optional;
+import org.assertj.core.api.OptionalAssert;
+
+final class InvocationResultAssert {
+
+    private InvocationResultAssert() {}
+
+    public static final class TaskResultOptionalAssert extends OptionalAssert<TaskResult> {
```

Comment:
Feels kinda annoying to have this nested inside another class - was the idea that there would be other assertions in here at some point?

Reply:
Can we just make this class top level now?